easypy/threadtree.py

```diff
@@ -170,27 +177,96 @@ def iter_thread_frames():
         yield from sys._current_frames().items()
 
 
-def walk_frames(thread=None, *, across_threads=False):
+class FrameWrapper:
+    """This wrapper adds the `f_thread_ident` attribute, which stores the thread
+    identifier of the current thread. All other attribute accesses are forwarded
+    to the wrapped frame object, making the wrapper behave like the original frame.
+    """
+
+    def __init__(self, frame, thread_ident):
+        self._frame = frame
+        self.f_thread_ident = thread_ident
+
+    def __getattr__(self, name):
+        """Delegate attribute access to the wrapped frame object."""
+        return getattr(self._frame, name)
+
+    def __setattr__(self, name, value):
+        """Allow setting attributes."""
+        if name == "_frame" or name == "f_thread_ident":
+            super().__setattr__(name, value)  # Handle internal attributes
+        else:
+            setattr(self._frame, name, value)  # Delegate to the wrapped frame
+
+    def __repr__(self):
+        """Provide a string representation including the class name and f_thread_ident."""
+        return (
+            f"<{self.__class__.__name__}(frame={repr(self._frame)}, "
+            f"f_thread_ident={self.f_thread_ident})>"
+        )
+
+    def __str__(self):
+        """Provide a readable string for the wrapper."""
+        return str(self._frame)
+
+    def __eq__(self, other):
+        """Equality comparison."""
+        if isinstance(other, self.__class__):
+            return (
+                self._frame == other._frame
+                and self.f_thread_ident == other.f_thread_ident
+            )
+        return self._frame == other
+
+    def __ne__(self, other):
+        """Inequality comparison."""
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        """Make the object hashable."""
+        return hash(self._frame)
+
+    def __dir__(self):
+        """List attributes of the wrapper and wrapped object."""
+        return dir(self._frame) + ["f_thread_ident"]
+
+
+def walk_frames(thread=None, *, across_threads=False, use_snapshots=False):
```

Comment:
I have put the change after a flag for backward compatibility

Reply:
I'm not really worried about backwards compatibility, especially since it's a useful/reliable behavior...  
The intention of `across_threads` has always been to get the original traceback. Let's drop the `use_snapshots` flag, make it the default behavior.